amethyst_rendy/src/submodules/vertex.rs

```diff
@@ -70,35 +169,53 @@ impl<B: Backend, T: 'static> DynamicVertex<B, T> {
         }
     }
 
+    /// Bind the allocated rendy buffer for this frame index.
     #[inline]
     pub fn bind(
         &self,
         index: usize,
         binding_id: u32,
+        offset: u64,
         encoder: &mut RenderPassEncoder<'_, B>,
     ) -> bool {
         self.per_image
             .get(index)
-            .map_or(false, |i| i.bind(binding_id, encoder))
+            .map_or(false, |i| V::bind(binding_id, encoder, &i.buffer, offset))
```

Comment:
it feels wrong to me that `binding_id` is being used only sometimes here. I think the whole `bind` method should be split in half and implemented only for relevant types. One for each impl of
```rs
impl<B: Backend, T: 'static> DynamicVertexData<B, VertexData<T>> { .. }
impl<B: Backend, T: 'static> DynamicVertexData<B, IndexData<T>> { .. }
```

Reply:
I've changed up the implementation now to specialize for each case and call the appropriate binding function, so there is no spare random value now.